File: src/_02_StructuralPattern/_02_05_Facade/java/after/EmailLib/MimeMessage.java
package _02_StructuralPattern._02_05_Facade.java.after.EmailLib;

import java.util.ArrayList;
import java.util.List;

public class MimeMessage {
  
  private Session session;
  private InternetAddress from;
  private List<InternetAddress> recipients;
  private String subject;
  private String text;

  public MimeMessage(Session session) {
    this.session = session;
    this.recipients = new ArrayList<InternetAddress>();
  }

  // session
  public Session getSession() {
    return session;
  }
  public void setSession(Session session) {
    this.session = session;
  }

  // from
  public InternetAddress getFrom() {
    return from;
  }
  public void setFrom(InternetAddress from) {
    this.from = from;
  }

  // recipients
  public List<InternetAddress> getRecipients() {
    return recipients;
  }
  public void addRecipient(InternetAddress recipient) {
    recipients.add(recipient);
  }

  // subject
  public String getSubject() {
    return subject;
  }
  public void setSubject(String subject) {
    this.subject = subject;
  }

  // text
  public String getText() {
    return text;
  }
  public void setText(String text) {
    this.text = text;
  }
  
}

File: src/_02_StructuralPattern/_02_05_Facade/EmailLib/MimeMessage.js
/**
 * @typedef { import("./Session").default } Session
 * @typedef { import("./InternetAddress").default } InternetAddress
 */

export default class MimeMessage {
  /** @type { Session } */
  #session;

  /** @type { InternetAddress } */
  #from;
  
  /** @type { InternetAddress[] } */
  #recipients;

  /** @type { string } */
  #subject;

  /** @type { string } */
  #text;

  /** @param { Session } session */
  constructor(session) {
    this.#session = session;
    this.#recipients = [];
  }

  // session
  /** @returns { Session } */
  getSession() {
    return this.#session;
  }
  /** @param { Session } session */
  setSession(session) {
    this.#session = session;
  }

  // from
  /** @returns { InternetAddress } */
  getFrom() {
    return this.#from;
  }
  /** @param { InternetAddress } from */
  setFrom(from) {
    this.#from = from;
  }

  // recipients
  /** @returns { InternetAddress[] } */
  getRecipients() {
    return this.#recipients;
  }
  /** @param { InternetAddress } recipient */
  addRecipient(recipient) {
    this.#recipients.push(recipient);
  }

  // subject
  /** @returns { string } */
  getSubject() {
    return this.#subject;
  }
  /** @param { string } subject */
  setSubject(subject) {
    this.#subject = subject;
  }

  // text
  /** @returns { string } */
  getText() {
    return this.#text;
  }
  /** @param { string } text */
  setText(text) {
    this.#text = text;
  }
}